### PR TITLE
automate vm_image_url and virtual_uc_ctlpane_interface 

### PR DESCRIPTION
**Pull request by Shrutikabansal**
update the vm_image_url based on the osp version and virtual_uc_ctlpane_interface based on the machine type

### CODE REVIEW

**Comment by masco** [comment]
Can one of the admins verify this patch?

File: group_vars/all.yml [hunk 1]
```diff
@@ -115,14 +115,13 @@ force_reprovision: false
 #This enables user to have a undercloud in VM
 #NOTE: now the vm is created in the ansible host, so
 #do not enable this while running from your desktop
-virtual_uc: false
+virtual_uc: true
```
Comment (asyedham) [inline group_vars/all.yml:118]: set to false

File: main.yml [hunk 2]
```diff
@@ -20,6 +25,12 @@
 - import_playbook: prepare_nic_configs.yml
   tags:
     - always
+  when: virtual_uc == false
+
+- import_playbook: prepare_nic_configs_virtual.yml
```
Comment (asyedham) [inline main.yml:30]: not needed. use the existing one as it seems just duplication of existing code

File: tasks/load_instackenv.yml [hunk 3]
```diff
@@ -31,6 +31,17 @@
   vars:
       instackenv_content: "{{ lookup('file', '{{ instackenv_file }}') | from_json }}"
 
+- name: skip first node of instack env file
```
Comment (asyedham) [inline tasks/load_instackenv.yml:34]: not needed. It was only for you to understand  the code :)

File: tasks/load_instackenv.yml [hunk 4]
```diff
@@ -31,6 +31,17 @@
...
+  set_fact:
+      instackenv_content: |
+          {% set a=instackenv_content.pop('nodes') %}
+          {{ instackenv_content | combine({'nodes': a|difference([a[0]])}, recursive=True) }}
+
+- name: update the instackenv file
+  copy:
+      dest: "{{ instackenv_file }}"
+      content: "{{ instackenv_content }}"
+
```
Comment (asyedham) [inline tasks/load_instackenv.yml:44]: not needed

**Review by asyedham — CHANGES_REQUESTED** [review]

**Comment by asyedham** [comment]
run tests

**Pull request closed without merging**